### ISSUE TITLE
EnqueueAsync call is ambiguous 

### ISSUE THREAD

**Issue by Jorixon** [CommunityToolkit/Windows#315]
### Describe the bug

It seems that using CommunityToolkit.WinUI.Extensions 8.0.240109 and CommunityToolkit.WinUI 7.1.2 causes EnqueueAsync to be an ambiguous reference. Thus I cannot build

### Steps to reproduce

```text
Create a WinUI app with the template: Blank App. Packaged (WinUl 3 in Desktop)
Update: Microsoft.Windows.SDK.BuildTools -> 10.0.22621.2428
 Microsoft.WindowsAppSDK -> 1.4.231219000

Then add:
CommunityToolkit.WinUI 7.1.2
CommunityToolkit.WinUI.Extensions 8.0.240109

This:
`DispatcherQueue.EnqueueAsync()`
Will now give an ambiguous reference error
```

### Expected behavior

That it does not produce the error

### Screenshots

![image](https://github.com/CommunityToolkit/Windows/assets/143650598/31421e00-2497-4996-ad83-13781d83134e)

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [X] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [X] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Visual Studio Version

2022

### Visual Studio Build Number

Version 17.8.3

### Device form factor

Desktop

### Additional context

This is problematic as I use a few libraries that are depended on CommunityToolkit.WinUI 7.1.2 so when I try to use any of the newer 8.* controls like CommunityToolkit.WinUI.Controls.RichSuggestBox I get this error.

### Help us help you

Yes, but only if others can assist.

**Comment by michael-hawker** [comment]
Yes, this is a known issue. 7.1 and 8.0 are mostly incompatible except for a few specific things that are independent like the Notifications and DataGrid.

What things are you still missing? Please feel free to search and vote on any missing controls you still are using.
